sound/soc/sof/intel/hda-loader.c

```diff
@@ -409,11 +410,12 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	if (i == HDA_FW_BOOT_ATTEMPTS) {
 		dev_err(sdev->dev, "error: dsp init failed after %d attempts with err: %d\n",
 			i, ret);
-		dev_err(sdev->dev, "ROM error=0x%x: FW status=0x%x\n",
+		dev_err(sdev->dev, "ROM error=0x%x: FW status=0x%x: PM status=0x%x\n",
```

Comment:
PM Status could be interpreted as device pm_runtime status, maybe make it clear that it's 'D0I3C status'

Also wondering how this was tested, because we don't support S0ix on any CI targets so far?

Reply:
@plbossart not tested in any way... relates to this: https://github.com/thesofproject/sof/issues/3267
I just tried to quickly dig out some power status registers (which I guess folks in this feature we're after).
You know this area probably much better than me, if you have some suggestions what should be logged 
here to get better understanding of the power state, let me know...

If I understood @lgirdwood wanted to have the power state of DSP and PCI (bus?), not the host, somehow visible.

Reply:
ok, but that specific register is only to signal D0i3, which can only happen *after* booting the firmware and will not help for resume or boot flows from D3.
In short I don't think this helps provide the diagnosis information @lgirdwood was asking for. I also don't understand how the DSP could be in D3 here, since it was typically powered before attempting to boot firmware. @ranj063 any ideas of what we could add here that might help triangulate such errors?

Reply:
@plbossart i'll need to comb through the docs to see what might be helpful here. But I agree VS_D0I3 bit is not useful during firmware boot at all.

Reply:
+ @yongzhi1 who reported the issue.

Reply:
@juimonen I would also expect to see any host side DSP PM register dumped here alongside the PCI PM register. This way @yongzhi1 will know for sure.

Reply:
@lgirdwood @juimonen I've added #2359 based on @mmaka1 's recommendation to dump the extended ROM status